src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,78 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    private static final String SUPPLY_CONDITION = "supply";
+    private static final String BUY_CONDITION = "buy";
+    private static final String REGEX_CONDITION = "\\D";
+    private StringBuilder stringBuilder = new StringBuilder();
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String textFromFile = readDataFromFile(fromFileName);
+        int[] array = calculateResultForDay(textFromFile);
+        String result = toStringForNewFile(array);
+        writeToFile(toFileName, result);
+
+    }
+
+    private String readDataFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = null;
+            value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find a file " + fromFileName + e);
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName + e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private int[] calculateResultForDay(String text) {
+        String[] allLines = text.split(System.lineSeparator());
+        int supplySum = 0;
+        int buySum = 0;
+        for (String line : allLines) {
+            allLines = line.split(",");
+            if (allLines [0].contains(SUPPLY_CONDITION)) {
+                supplySum += Integer.parseInt(line.replaceAll(REGEX_CONDITION, ""));
+            }
+            if (allLines [0].contains(BUY_CONDITION)) {
+                buySum += Integer.parseInt(line.replaceAll(REGEX_CONDITION, ""));
+            }
+        }
+        return new int[]{buySum, supplySum, supplySum - buySum};
+    }
+
+    private String toStringForNewFile(int[] results) {
```

Comment:
```suggestion
    private String createReport(int[] results) {
```
use verbs for method names